manifests/00-custom-resource-definition.yaml

```diff
@@ -280,7 +281,7 @@ spec:
                                     format: duration
                                     type: string
                                 type: object
-                              networkLoadBalancer:
+                              nlbParameters:
```

Comment:
I can see `properties:eipAllocations:` in the API PR, should we add same here?

Reply:
Originally we had [this](https://github.com/openshift/cluster-ingress-operator/blob/master/hack/update-generated-crd.sh#L15)  but after the feature gate changes in the API,  that file is no longer present so I modified the hack to have 

```
# Can't rely on associative arrays for old Bash versions (e.g. OSX)
install_crd \
  "vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-Default.crd.yaml" \
  "manifests/00-custom-resource-definition.yaml"
```

However the `vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-Default.crd.yaml` does not have `eipAllocations` in it which might because I added featuregates on it
[operator/v1 CRD](https://github.com/openshift/api/pull/1826/commits/24253ce8e0ca5da2161ba0ea4abc7f5659b8ae4c#diff-891d2cdcb01603ef2a2c1bfb2f6ef32cf2acf8b1471871a5a6e27308d50050f9R645)
[config/v1 CRD](https://github.com/openshift/api/pull/1826/commits/24253ce8e0ca5da2161ba0ea4abc7f5659b8ae4c#diff-0beb33df12dd4ecac8d28c3fe688acc3a178de15c301325e33f586207b2c748cR171)

[Enabled in](https://github.com/openshift/api/pull/1826/commits/24253ce8e0ca5da2161ba0ea4abc7f5659b8ae4c#diff-a8b6135d50534471326ea7bcd20e0f5eae25353f7788338060f718128a6a0b34R78) 


In the `0000_50_ingress_00_ingresscontrollers-Default.crd.yaml` file I don't see `eipAllocations` being added.
```
misalunk@misalunk-mac cluster-ingress-operator % cat vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-Default.crd.yaml  | grep eipAllocation 
misalunk@misalunk-mac cluster-ingress-operator %
```

However, in the following files I do see them getting added.
```
misalunk@misalunk-mac cluster-ingress-operator % cat vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-DevPreviewNoUpgrade.crd.yaml  | grep eipAllocation
                                  eipAllocations:
                                  eipAllocations:
misalunk@misalunk-mac cluster-ingress-operator % 

misalunk@misalunk-mac cluster-ingress-operator % cat vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-TechPreviewNoUpgrade.crd.yaml | grep eipAllocation
                                  eipAllocations:
                                  eipAllocations:
misalunk@misalunk-mac cluster-ingress-operator % 
misalunk@misalunk-mac cluster-ingress-operator % 
misalunk@misalunk-mac cluster-ingress-operator % cat vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-CustomNoUpgrade.crd.yaml | grep eipAllocation
                                  eipAllocations:
                                  eipAllocations:
misalunk@misalunk-mac cluster-ingress-operator %
```

@Miciah do you think we shall change the filename present in [update-generated-crd.sh](https://github.com/openshift/cluster-ingress-operator/blob/master/hack/update-generated-crd.sh#L15) to any one of the above 3 files which have `eipAllocations` available ?

Reply:
I believe `update-generated-crd.sh` needs to copy all CRD variants from `vendor/` to `manifests/`.  CVO should figure out which one of the variants to use based on their respective `release.openshift.io/feature-set` annotations.

Reply:
Also, please note that you cannot change the name of a field that already exists in shipped releases, as I noted here: https://github.com/openshift/api/pull/1826#discussion_r1581480212.

Reply:
The only difference is the between `vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-Default.crd.yaml `. and the above mentioned 3 files is that it does not. have eipAllocations as mentioned in the following:
```
misalunk@misalunk-mac cluster-ingress-operator % diff vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-CustomNoUpgrade.crd.yaml vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-Default.crd.yaml 
10c10
<     release.openshift.io/feature-set: CustomNoUpgrade
---
>     release.openshift.io/feature-set: Default
288,300d287
<                                 properties:
<                                   eipAllocations:
<                                     description: 'You can assign Elastic IP addresses
<                                       to the Network Load Balancer by adding the following
<                                       annotation. The number of Allocation IDs must
<                                       match the number of subnets that are used for
<                                       the load balancer. service.beta.kubernetes.io/aws-load-balancer-eip-allocations:
<                                       eipalloc-xxxxxxxxxxxxxxxxx,eipalloc-yyyyyyyyyyyyyyyyy
<                                       https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
<                                       https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.7/guide/service/annotations/#eip-allocations'
<                                     items:
<                                       type: string
<                                     type: array
1945,1957d1931
<                                 properties:
<                                   eipAllocations:
<                                     description: 'You can assign Elastic IP addresses
<                                       to the Network Load Balancer by adding the following
<                                       annotation. The number of Allocation IDs must
<                                       match the number of subnets that are used for
<                                       the load balancer. service.beta.kubernetes.io/aws-load-balancer-eip-allocations:
<                                       eipalloc-xxxxxxxxxxxxxxxxx,eipalloc-yyyyyyyyyyyyyyyyy
<                                       https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
<                                       https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.7/guide/service/annotations/#eip-allocations'
<                                     items:
<                                       type: string
<                                     type: array
misalunk@misalunk-mac cluster-ingress-operator % 
misalunk@misalunk-mac cluster-ingress-operator % 
misalunk@misalunk-mac cluster-ingress-operator % 
misalunk@misalunk-mac cluster-ingress-operator % diff vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-TechPreviewNoUpgrade  vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-Default.crd.yaml
diff: vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-TechPreviewNoUpgrade: No such file or directory
misalunk@misalunk-mac cluster-ingress-operator % diff vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-TechPreviewNoUpgrade.crd.yaml  vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-Default.crd.yaml  
10c10
<     release.openshift.io/feature-set: TechPreviewNoUpgrade
---
>     release.openshift.io/feature-set: Default
288,300d287
<                                 properties:
<                                   eipAllocations:
<                                     description: 'You can assign Elastic IP addresses
<                                       to the Network Load Balancer by adding the following
<                                       annotation. The number of Allocation IDs must
<                                       match the number of subnets that are used for
<                                       the load balancer. service.beta.kubernetes.io/aws-load-balancer-eip-allocations:
<                                       eipalloc-xxxxxxxxxxxxxxxxx,eipalloc-yyyyyyyyyyyyyyyyy
<                                       https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
<                                       https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.7/guide/service/annotations/#eip-allocations'
<                                     items:
<                                       type: string
<                                     type: array
1945,1957d1931
<                                 properties:
<                                   eipAllocations:
<                                     description: 'You can assign Elastic IP addresses
<                                       to the Network Load Balancer by adding the following
<                                       annotation. The number of Allocation IDs must
<                                       match the number of subnets that are used for
<                                       the load balancer. service.beta.kubernetes.io/aws-load-balancer-eip-allocations:
<                                       eipalloc-xxxxxxxxxxxxxxxxx,eipalloc-yyyyyyyyyyyyyyyyy
<                                       https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
<                                       https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.7/guide/service/annotations/#eip-allocations'
<                                     items:
<                                       type: string
<                                     type: array
misalunk@misalunk-mac cluster-ingress-operator % 
misalunk@misalunk-mac cluster-ingress-operator % 
misalunk@misalunk-mac cluster-ingress-operator % 
misalunk@misalunk-mac cluster-ingress-operator % diff vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-DevPreviewNoUpgrade.crd.yaml  vendor/github.com/openshift/api/operator/v1/zz_generated.crd-manifests/0000_50_ingress_00_ingresscontrollers-Default.crd.yaml  
10c10
<     release.openshift.io/feature-set: DevPreviewNoUpgrade
---
>     release.openshift.io/feature-set: Default
288,300d287
<                                 properties:
<                                   eipAllocations:
<                                     description: 'You can assign Elastic IP addresses
<                                       to the Network Load Balancer by adding the following
<                                       annotation. The number of Allocation IDs must
<                                       match the number of subnets that are used for
<                                       the load balancer. service.beta.kubernetes.io/aws-load-balancer-eip-allocations:
<                                       eipalloc-xxxxxxxxxxxxxxxxx,eipalloc-yyyyyyyyyyyyyyyyy
<                                       https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
<                                       https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.7/guide/service/annotations/#eip-allocations'
<                                     items:
<                                       type: string
<                                     type: array
1945,1957d1931
<                                 properties:
<                                   eipAllocations:
<                                     description: 'You can assign Elastic IP addresses
<                                       to the Network Load Balancer by adding the following
<                                       annotation. The number of Allocation IDs must
<                                       match the number of subnets that are used for
<                                       the load balancer. service.beta.kubernetes.io/aws-load-balancer-eip-allocations:
<                                       eipalloc-xxxxxxxxxxxxxxxxx,eipalloc-yyyyyyyyyyyyyyyyy
<                                       https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
<                                       https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.7/guide/service/annotations/#eip-allocations'
<                                     items:
<                                       type: string
<                                     type: array
misalunk@misalunk-mac cluster-ingress-operator % 
```

Reply:
Or i think best is to add configv1.Default to https://github.com/openshift/api/pull/1826/commits/24253ce8e0ca5da2161ba0ea4abc7f5659b8ae4c#diff-a8b6135d50534471326ea7bcd20e0f5eae25353f7788338060f718128a6a0b34R78